futlib/math.fut

```diff
@@ -394,7 +394,7 @@ module i64: (size with t = i64) = {
   let num_bits = 64
   let get_bit (bit: i32) (x: t) = to_i32 ((x >> i32 bit) & i32 1)
   let set_bit (bit: i32) (x: t) (b: i32) =
-    ((x & i32 (intrinsics.~(1 intrinsics.<< b))) | x intrinsics.<< i32 b)
+    ((x & i32 (intrinsics.~(1 intrinsics.<< bit))) | i32 (b intrinsics.<< bit))
```

Comment:
Try making the RHS a conversion from `u32` instead.

Reply:
The shadowing inside the module makes it really obnoxious to write.  You want something like:

    ((x & i32 (intrinsics.~(1 intrinsics.<< bit))) | intrinsics.zext_i32_i64 (b intrinsics.<< bit))